src/nix/fmt.cc

```diff
@@ -0,0 +1,73 @@
+#include "command.hh"
+#include "run.hh"
+
+using namespace nix;
+
+struct CmdFmt : SourceExprCommand {
```

Comment:
This should require the experimental flakes feature or support non-flakes use cases.

Reply:
It will ask for the flakes flag before getting there

Other commands have the same behaviour and UX

```
[kamadorueda@nixos:/data/nix]$ ./outputs/out/bin/nix fmt
error: experimental Nix feature 'nix-command' is disabled; use '--extra-experimental-features nix-command' to override

[kamadorueda@nixos:/data/nix]$ ./outputs/out/bin/nix --extra-experimental-features nix-command fmt
error: you must pass '--formatter <flake-uri>'
Try './outputs/out/bin/nix --help' for more information.

[kamadorueda@nixos:/data/nix]$ ./outputs/out/bin/nix --extra-experimental-features nix-command fmt --formatter nixpkgs#nixpkgs-fmt
error: experimental Nix feature 'flakes' is disabled; use '--extra-experimental-features flakes' to override

[kamadorueda@nixos:/data/nix]$ ./outputs/out/bin/nix --extra-experimental-features nix-command --extra-experimental-features flakes fmt --formatter nixpkgs#nixpkgs-fmt
(works)
```

It's intentional to make it work with flakes only, as other commands like `nix bundle` do

There is also a warning about experimental stuff:

![image](https://user-images.githubusercontent.com/47480384/153652760-0052718d-4b81-462f-ac38-b445e3b17bee.png)

or maybe I'm not understanding your suggestion, could you please help me understand?